crates/cast/src/starknet_commands/script.rs

```diff
@@ -243,6 +243,62 @@ impl CairoHintProcessor<'_> {
 
                 Ok(())
             }
+            "deploy" => {
+                let class_hash = inputs[0].to_field_element();
+                let calldata_length = inputs[1]
+                    .to_usize()
+                    .expect("Failed to convert calldata length to usize");
+                let constructor_calldata: Vec<FieldElement> = {
+                    let calldata = Vec::from(&inputs[2..(2 + calldata_length)]);
+                    calldata
+                        .iter()
+                        .map(StarknetConversions::to_field_element)
+                        .collect()
+                };
+                let mut offset = 2 + calldata_length;
+                let salt = if inputs[offset] == 0.into() {
+                    offset += 1;
+                    Some(inputs[offset].to_field_element())
+                } else {
+                    None
+                };
+                offset += 1;
+                let unique = { inputs[offset] == 1.into() };
+                offset += 1;
+                let max_fee = if inputs[offset] == 0.into() {
+                    offset += 1;
+                    Some(inputs[offset].to_field_element())
+                } else {
+                    None
+                };
```

Comment:
These can be done like the following:
https://github.com/foundry-rs/starknet-foundry/blob/3db4ee2c3fc4a04546da7e607c513588acbb32af/crates/forge-runner/src/test_execution_syscall_handler.rs#L341-L344

Reply:
that is cool, but we also have calldata of various length to consider. I agree this could be refactored, but I'm not sure this will be that straightforward. Let me save this one and get back to it after release :+1: